File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/format/CentsToAmount.java
package com.freight.format;

/** Formatter: cents to amount. */
public final class CentsToAmount implements Formatter {

    private static final String BASE36_DIGITS = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    @Override
    public String name() {
        return "cents_to_amount";
    }

    @Override
    public String apply(long value) {
        boolean negative = value < 0L;
        long absolute = negative ? -value : value;
        return (negative ? "-" : "") + (absolute / 100L) + "."
                + String.format(java.util.Locale.ROOT, "%02d", absolute % 100L);
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/format/cents_to_amount.cpp
#include "freight/formats.h"

#include <algorithm>
#include <cstdio>

namespace freight {

// Formatter: cents to amount.
std::string fmt_cents_to_amount(long long value) {
  bool negative = value < 0;
  long long absolute = negative ? -value : value;
  char buffer[64];
  std::snprintf(buffer, sizeof(buffer), "%s%lld.%02lld", negative ? "-" : "", absolute / 100,
                absolute % 100);
  return std::string(buffer);
}

}  // namespace freight
